### PR TITLE
Feature/tao 10365 apply access control to queries

### PR DESCRIPTION
**Pull request by luisfn**
With this PR, we are adding the list of roles the logged user have to the elastic search query, in order to only bring results that he have access to see.

### CODE REVIEW

File: src/QueryBuilder.php [hunk 1]
```diff
@@ -49,8 +56,8 @@ public static function create(): self
 
     public function getSearchParams(string $queryString, string $type, int $start, int $count, string $order, string $dir): array
     {
-        $qs = htmlspecialchars_decode($queryString);
-        $blocks = $output = preg_split( '/( AND )/', $qs);
+        $decoded_query_string = htmlspecialchars_decode($queryString);
```
Comment (jbout) [inline src/QueryBuilder.php:59]: Is this the correct place to decode the string?

Comment (luisfn) [reply src/QueryBuilder.php:59]: I think it's ok to be here or I can move it to ElasticSearch->query() method that calls it, but it won't change much I think

File: src/QueryBuilder.php [hunk 2]
```diff
@@ -117,4 +128,24 @@ private function buildCustomConditions(string $fieldName, string $term): string
...
+        $current_user = $this->getServiceLocator()->get(SessionService::SERVICE_ID)->getCurrentUser();
+
+        foreach ($current_user->getRoles() as $role) {
+            $conditions[] = sprintf('%s:"%s"', self::READ_ACCESS_FIELD, $role);
+        }
+
+        return '(' . implode(' OR ', $conditions). ')';
+    }
+
+    private function includeAccessData(): bool {
```
Comment (martijn-tao) [inline src/QueryBuilder.php:146]:
```suggestion
    private function includeAccessData(): bool
    {
```

File: test/QueryBuilderTest.php [hunk 3]
```diff
@@ -68,7 +80,74 @@ public function queryResults() {
...
+            ],
+            'body' => $body,
+        ];
+
+        $result = $this->subject->getSearchParams($queryString, TaoOntology::CLASS_URI_ITEM, 0, 10, '_id', 'DESC');
+
+        $this->assertSame($expected, $result);
+    }
+
+    public function queryResultsWithAccessControl() {
```
Comment (Skyld) [inline test/QueryBuilderTest.php:116]: Return type hint is missing

**Pull request merged**